packages/server/db/users.js

```diff
@@ -32,15 +105,22 @@ const parseMentor = async (body) => {
     tags: body.tags,
     subjects: body.subjects,
     gradeLevels: body.gradeLevels,
-    timzone: body.timezone,
+    timezone: body.timezone,
   };
 
-  // TODO: validate data
+  /**
+   * The validation asynchronously checks the data against the restrictions of Yup
+   * TODO: More validation needs to be done on how the error is returned to the server.
+   */
+  async function validate() {
+    const valid = await mentorSchema.isValid(mentor);
+  }
+  validate().then(() => true);
```

Comment:
Since you made parseMentor an async method, do you need create this inner validate function?

It should be possible to call `await mentorSchema.isValid(mentor);`